lib/modules/comfy_opengraph.rb

```diff
@@ -29,21 +31,62 @@ def parse(opengraph: nil, page: nil)
       # set opengraph meta-tag via a new hash
       payload = {}
       payload[@mappings[id]] = value
-      opengraph.content('og', payload)
+      opengraph.content(type, payload)
     end
   end
 
   private
 
   def get_fragment_value(fragment)
     if fragment.tag =~ /file/ && fragment.attachments.first # get path when fragment is file
-      if Rails.env.development?
-        URI.join(root_url, rails_blob_path(fragment.attachments.first.blob, only_path: true))
-      else
-        fragment.attachments.first.service_url&.split('?')&.first
-      end
-    else # expect a string by default
+      Rails.env.development? ? local_url(fragment) : production_url(fragment)
+    else 
+      process_meta_tags(fragment)
+    end
+  end
+
+  def local_url(fragment)
+    URI.join(root_url, rails_blob_path(fragment.attachments.first.blob, only_path: true))
+  end
+
+  def production_url(fragment)
+    fragment.attachments.first.service_url&.split('?')&.first
+  end
+
+  def process_meta_tags(fragment)
+    identifier = fragment.identifier
+    
+    case identifier
+    when 'social-title'
+      return og_title
+    when 'social-description'
+      return og_description
+    when 'image'
+      return og_image
+    else
+      # expect a string by default
       fragment.content&.squish
     end
   end
+
+  def og_description
+    social_desc = cms_fragment_content(:social_description, @page)
+    summary = cms_fragment_content(:summary, @page)
+    fallback_summary = summary.blank? ? I18n.t('meta.site.description') : summary
+    social_desc.blank? ? fallback_summary : social_desc
+  end
+
+  def og_image
+    image = cms_fragment_content(:image, @page).try(:attachments)&.first
+    path_to_image = URI.join(root_url, url_for(image))
```

Comment:
If `image` is nil this line would probably fail or return an undesired behaviour